### PR TITLE
Feature/resource status

### PR DESCRIPTION
**Pull request by aclassen**

### CODE REVIEW

File: core/src/main/java/io/stanwood/framework/arch/core/Resource.kt [hunk 1]
```diff
@@ -24,5 +24,5 @@ package io.stanwood.framework.arch.core
 sealed class Resource<out T>(open val data: T? = null) {
     data class Success<out T>(override val data: T) : Resource<T>()
     class Failed<out T>(val msg: String, val cause: Throwable? = null, data: T? = null) : Resource<T>(data)
-    class Loading<out T> : Resource<T>()
+    class Loading<out T>(data: T? = null) : Resource<T>(data)
```
Comment (ubuntudroid) [inline core/src/main/java/io/stanwood/framework/arch/core/Resource.kt]: Nice, that should remove the need to cast to access data in some places.

File: core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt [hunk 2]
```diff
@@ -44,25 +44,27 @@ object ResourceStatusTransformer {
...
-            upstream.map { toStatus(it) }
+            upstream.map { it.toResourceStatus() }
     }
-}
\ No newline at end of file
+}
+
+fun <T> Resource<T>.toResourceStatus() =
+    data
+        ?.let { ResourceStatus.Success }
```
Comment (ubuntudroid) [inline core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt]: Not sure I understand this (I know, it has been like this before, but I only now recognized that I in fact don't understand this :D ) - why does a non-null data always result in a success? Given the current resource definition the other states also *may* have a non-null data.

File: core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt [hunk 3]
```diff
@@ -44,25 +44,27 @@ object ResourceStatusTransformer {
...
     private class SingleResourceTransformer<T> : SingleTransformer<Resource<T>, ResourceStatus> {
         override fun apply(upstream: Single<Resource<T>>): SingleSource<ResourceStatus> =
-            upstream.map { toStatus(it) }
+            upstream.map { it.toResourceStatus() }
     }
-}
\ No newline at end of file
+}
+
+fun <T> Resource<T>.toResourceStatus() =
```
Comment (ubuntudroid) [inline core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt]: General question: what is `ResourceStatus` used for?

Comment (aclassen) [reply core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt]: To bind it to your ui and show loding,success,error state

File: core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt (hunk 2 above)
Comment (aclassen) [reply core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt]: You are right , i need to rework this in next version.

File: core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt (hunk 3 above)
Comment (ubuntudroid) [reply core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt]: So a simplified Resource without the data, right?

**Pull request merged**